great_expectations/data_context/data_context.py

```diff
@@ -1123,6 +1203,344 @@ def run_validation_operator(
                 **kwargs,
             )
 
+    def _get_data_context_version(self, arg1: Any, **kwargs) -> Optional[str]:
+        """
+        arg1: the first positional argument (can take on various types)
+
+        **kwargs: variable arguments
+
+        Returns None if no datasources have been configured (or if there is an exception while getting the datasource).
+        Returns "v3" if the datasource is a subclass of the BaseDatasource class.
+        Returns "v2" if the datasource is an instance of the LegacyDatasource class.
+        """
+
+        if not self.datasources:
+            return None
+
+        if {
+            "datasource_name",
+            "data_connector_name",
+            "data_asset_name",
```

Comment:
This one seems dangerous -- didn't data_asset_name also appear in the legacy request? If not, disregard.

Reply:
@jcampbell In the Legacy API, `data_asset_name` is not part of the `get_batch()` method (i.e, `_get_batch_v2()` to be exact, in the present PR).  It can be built into `batch_kwargs`, but it does not directly participate in getting the batch (and there is no `**kwargs` parameter in `_get_batch_v2()`) -- so I believe we are safe.